src/ReferenceRepository/DataObjectReferenceRepository.php

```diff
@@ -0,0 +1,77 @@
+<?php declare(strict_types=1);
+
+namespace Neusta\Pimcore\FixtureBundle\ReferenceRepository;
+
+/**
+ * based on \Doctrine\Common\DataFixtures\ReferenceRepository.
+ *
+ * @template T of object
+ */
+class DataObjectReferenceRepository
```

Comment:
I wonder if we should rename this to `ElementReferenceRepository` or maybe `PimcoreElementReferenceRepository`, since you may not just create `DataObject` fixtures, but maybe `Document` or `Asset` fixtures as well.

Reply:
The ReferenceRepository shipped with the doctrine-fixtures-bundle provides functionality which only works with Doctrine-Entities. This Reference-Repository does not care about "what" object you try to store and could be renamed to `ObjectReferenceRepository` or just `ReferenceRepository` as it is not Pimcore specific.